test/gitlab.js

```diff
@@ -89,7 +89,9 @@ test.serial('should upload assets and release', async t => {
   interceptPublish({
     body: {
       name: 'Release 2.0.1',
+      ref: 'feature/gitlab-release-with-ref',
```

Comment:
This is causing a unit test problem since `branchName` resolves differently when tests are run locally than when being run on the server.  Locally this resolves to `'feature/gitlab-release-with-ref'` but it is resolving to `'HEAD'` when run on GitHub.

Reply:
@webpro, I think I've reached the limit of how I can help here without some guidance.  I'm having a heckuva time trying to figure out how to solve this.  There's very little inline documentation, and I'm unfamiliar with several of the core libraries being used.

Reply:
Unless I'm mistaken, one of the key differences in the `gitlab.js` tests is that it does not start by initializing an empty git repository from which to run tests.  Both `tasks.interactive.js` and `tasks.js` do this with `beforeEach()` but `gitlab.js` does not.  Because of this, the tests in this suite are not deterministic because `branchName` would evaluate differently for every branch in which the tests are run.  This feels like a more fundamental problem with the unit test setup than I am equipped to resolve and somewhat outside the scope of this specific ticket.

Reply:
One thing we _could_ do here to side-step the issue entirely is to change the `nock` body matching for the `ref` property to some type of regex for that specific test:
![image](https://github.com/user-attachments/assets/afb44748-cec1-4a72-8a58-09105aabe162)
I don't know how you'd feel about that.  It feels a little like cheating. 🙂

Reply:
Wouldn't be the end of the world if required, but let's try my suggestion first :)

Reply:
Which suggestion?  This definitely won't continue working as-is because I currently have the name of my current branch in there.  If I were to leave it as-is, it would eventually fail on someone else's branch with a different name.  One other thing we could do is detect the current branch's name and use that in lieu of my hard-coded one in your `isCI` check.